vernac/g_vernac.mlg

```diff
@@ -108,34 +108,43 @@ GRAMMAR EXTEND Gram
     ]
   ;
   vernac:
-    [ [ IDENT "Local"; v = vernac_poly -> { let (f, v) = v in (("local", VernacFlagEmpty) :: f, v) }
-      | IDENT "Global"; v = vernac_poly -> { let (f, v) = v in (("global", VernacFlagEmpty) :: f, v) }
-
-      | v = vernac_poly -> { v } ]
-    ]
-  ;
-  vernac_poly:
-    [ [ IDENT "Polymorphic"; v = vernac_aux ->
+    [ [ IDENT "Local"; v = vernac ->
```

Comment:
>Should there be a check for inconsistent attributes, e.g. if both "Global" and "Local" are present? On the other hand, these are legacy attributes and the current syntax doesn't permit this case. No strong opinion on this.

Through this change, we move from a check at parsing time to a check at attribute parsing time (which is done as a separate phase). Global + Local are rejected like so:

```
Local Global Variable foo : Prop.
Error: Attribute for Locality specified twice.
```

> Also, would it make sense to create a "legacy_attr" nonterminal and have the single production `vernac ::= LIST0 legacy_attr vernac_aux`? There's a fair bit of repeated code here.

I can try, but my previous attempt at doing this change (as requested by @SkySkimmer) was unsuccessful (cf. https://github.com/coq/coq/pull/11665#discussion_r383364903) and I'm afraid I will encounter again the same issue.

Reply:
Regarding `legacy_attr`, how about asking our parser experts what's going on?  As far as I can see, this is not ambiguous.  I don't see why it causes a problem.  And if it's a fixable bug, we should do so.

It's a bit of a mystery exactly what grammars work and don't work.  I have some idea about this but not a complete understanding.  This would be an excellent thing to document (and in the not distant future?), perhaps in the Coq manual.  It would help developers and plugin writers.

Reply:
Who knows, maybe it will work if the the regular `#[ ... ]` attributes are in separate productions.

Reply:
Apparently, it works. That makes things much simpler.

Reply:
Cool.

If you want the doc to group some of the alternatives on one line such Local/Global shown at the bottom then the easiest way to do that is in the mlg (provided the parser is OK with it), e.g.

```
  legacy_attr:
    [ [   [ "Local" -> { <code> } | "Global" -> { <code> } ];
       :
```
Otherwise you have to add edits in common.edit_mlg, as for assumption_token

```
   legacy_attr ::= {| Local | Global }
   | {| Polymorphic | Monomorphic }
   | {| Cumulative | NonCumulative }
   | Private
   | Program
```

Reply:
I'd rather not edit the mlg for something that is purely about rendering the doc.

Reply:
We should reduce the 1500 lines of edits in common.edit_mlg where possible.  The relationship between the grammar in the rsts and in the mlgs can be pretty non-obvious.  I've gotten reasonably good at it but you have to know exactly how the edits interact.  Doing it in the mlg saves 6 lines of editing.

Reply:
Yes, I agree.

Reply:
What you suggest isn't valid `mlg` though.